### DISCUSSION TITLE
How to handle edge bypassing a node

### DISCUSSION THREAD

**Discussion by RichestYoo** [xyflow/xyflow#2400] [Ideas]
Hello.

I'm working in AI area and one of classical model in this area is called ResNet which looks like this:
![image](https://user-images.githubusercontent.com/51534815/188258255-4d1b80f3-2baa-4cb5-9d53-1afb4c5b42c2.png)

When drawing this in a typical code from ReactFlow, edge going to the right side (named "Identity") goes down at the center and thereby overlaps with other edges. When handling a simple model, I can redesign the view by making the right line at the center and put the nodes at the right, but when the model gets complex, it's tough to rearrange whole model.

Do you have some way where 1) those overlapping edges avoid collision from one another or 2) some way to make an edge follow a path user designates? I found other tools having part of them in the hope that you may check its possibility & feasibility quickly:
1) https://mrblenny.github.io/react-flow-chart/index.html?path=/story/other-config--smart-link-routing 
   (decided not to use this because it's been left unchanged for two years, meaning React upgrade will cause problems in the future)
2) https://github.com/projectstorm/react-diagrams  whose test page is https://projectstorm.cloud/react-diagrams/?path=/story/simple-usage--canvas-drag : you can make dots in the middle of an edge, just like you edit points in Powerpoint shapes.

Thank you.

**Comment by imabp** [comment]
@RichestYoo did you find any solution to this? Stuck in a similar situation. 
The last resort is to bypass reactflow grid system and create local grid system and use A* algorithm

**Reply by tisoap** [reply]
I've done this in this library: [react-flow-smart-edge](https://github.com/tisoap/react-flow-smart-edge)

**Reply by imabp** [reply]
Nice, didn't know that.But I have built out this custom solution it's working so far well since 8 months 🙈. Thanks for sharing the library

**Comment by tisoap** [comment]
I'm the creator of [react-flow-smart-edge](https://github.com/tisoap/react-flow-smart-edge) , a library that uses using A* pathfinding to route edges around other nodes, avoiding intersections. It has smart equivalents for every type of edge React Flow has. It's been updated to work with React Flow 12.
